File: src/BaseHmiTypes/Screens/Base/HmiFont.cs
namespace BaseHmiTypes.Screens.Base;

public class HmiFont
{
    public HmiProperty<string>? Name { get; set; }

    public HmiProperty<double>? Size { get; set; }

    public HmiProperty<double>? CharacterWidth { get; set; }

    public HmiProperty<double>? EscapementAngle { get; set; }

    public HmiProperty<double>? OrientationAngle { get; set; }

    public HmiProperty<int>? Weight { get; set; }

    public HmiProperty<bool>? Bold { get; set; }

    public HmiProperty<bool>? Italic { get; set; }

    public HmiProperty<bool>? Underline { get; set; }

    public HmiProperty<bool>? Strikethrough { get; set; }

    public HmiProperty<int>? CharacterSet { get; set; }

    public HmiProperty<int>? OutputPrecision { get; set; }

    public HmiProperty<int>? ClippingPrecision { get; set; }

    public HmiProperty<int>? Quality { get; set; }

    public HmiProperty<int>? PitchAndFamily { get; set; }
}

File: typescript/base-hmi-types/src/screens/base/HmiFont.ts
import { HmiProperty } from "./HmiProperty.js";

export class HmiFont {
  name?: HmiProperty<string>;
  size?: HmiProperty<number>;
  characterWidth?: HmiProperty<number>;
  escapementAngle?: HmiProperty<number>;
  orientationAngle?: HmiProperty<number>;
  weight?: HmiProperty<number>;
  bold?: HmiProperty<boolean>;
  italic?: HmiProperty<boolean>;
  underline?: HmiProperty<boolean>;
  strikethrough?: HmiProperty<boolean>;
  characterSet?: HmiProperty<number>;
  outputPrecision?: HmiProperty<number>;
  clippingPrecision?: HmiProperty<number>;
  quality?: HmiProperty<number>;
  pitchAndFamily?: HmiProperty<number>;
}
